packages/lit-dev-content/site/tutorials/content/tooltip/08.md

```diff
@@ -0,0 +1,74 @@
+The tooltip supports good performance via lazy creation, but using the
+`lazy` API is cumbersome. In addition, as implemented, the tooltip content
+doesn't update when the greeted person's name changes.
+
+A Lit directive can address these issues by encapsulating the tooltip's `lazy`
+API. Users of the directive will use an
+[element expression](http://lit.dev/docs/templates/expressions/#element-expressions)
+on the target element and provide a
+[TemplateResult](http://lit.dev/docs/libraries/standalone-templates/#rendering-lit-html-templates)
+containing the tooltip content.
+
+A scaffold for the directive is setup for you to fill in with the `update`
+method already completed. It stores directive `part` and the `tooltipContent`
+passed in so you can use them in the methods you'll add: `setupLazy` and
+`renderTooltipContent`.
+
+```ts
+update(part: ElementPart, [tooltipContent]: DirectiveParameters<this>) {
+  this.tooltipContent = tooltipContent;
+  this.part = part;
+  if (!this.didSetupLazy) {
+    this.setupLazy();
+  }
+  if (this.tooltip) {
+    this.renderTooltipContent();
+  }
+}
+```
+
+First add the `setupLazy` method to the directive. It should set `didSetupLazy`
+so that it's only called once and then call `SimpleTooltip.lazy`. The call to
+`lazy` will store the `tooltip` on the directive and then call the
+`renderTooltipContent` method that you'll implement next.
+
+```ts
+setupLazy() {
+  this.didSetupLazy = true;
+  SimpleTooltip.lazy(this.part!.element, (tooltip: SimpleTooltip) => {
+    this.tooltip = tooltip;
+    this.renderTooltipContent();
+  });
+}
+```
+
+Now add the `renderTooltipContent` method to the directive. It will call Lit's
+`render` function to render the `tooltipContent` into the tooltip element.
+You'll use the part's `renderOptions` so they stay in sync.
+
+```ts
+renderTooltipContent() {
+  render(this.tooltipContent, this.tooltip!, this.part!.options);
+}
+```
+
+Now open the `my-content.ts` module. That cumbersome code you previously added
+to call `lazy` in `firstUpdated` has been removed and the `tooltip` directive
+has been imported. Find the greeting and add the directive.
+
+```html
+<p>
+  <span ${tooltip(html`${this.name}, there's coffee available in the lounge.`)}>
+    Hello, ${this.name}!
+  </span>
+</p>
+```
+
+<aside class="info">You're just rendering text here, but since the directive
+takes any value Lit can render, it could include rich HTML elements like images
+or tables.</aside>
+
+Finally, interact with the greeting to verify that the tooltip works.
```

Comment:
I'd implore you to add a congrats step!

before can be the dumb tooltip and after can be this beautiful directified tooltip.

Also tell the user what concepts and docs to read more on that would be related to this or what to learn from here

Reply:
Done